apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/consumer/ConsumeDriver.java

```diff
@@ -91,7 +91,7 @@ public void begin(Channels channels) {
     }
 
     @Override
-    public boolean isRunning(Channels channels) {
+    public synchronized boolean isRunning(Channels channels) {
         return running;
     }
```

Comment:
I think that to change to a `volatile` variable is enough. I mean replacing `boolean running` with `volatile boolean running`.

Reply:
ok